docs/language-spec.md

```diff
@@ -2082,6 +2096,9 @@ interpreted as an open range (i.e. inclusive) where the start and end values are
 
 Reserved ranges for the same enum are not allowed to overlap one another.
 
+The contents of the string literal that defines a reserved name must represent a valid
+identifier. In other words, they must match the production for the
+[_identifier_](#identifiers-and-keywords) lexical element.
```

Comment:
Am I correct in understanding that this check in protoc is just a warning for now, or did it change since the PR?

Reply:
You are correct, it is a warning in `protoc`. But it will be promoted to an error in the following release.

It was agreed that the existing behavior was incorrect, so I think it's okay for this spec to vary from actual `protoc` behavior if it is an acknowledged defect in `protoc`. In fact, even [their spec](https://developers.google.com/protocol-buffers/docs/reference/proto3-spec#reserved) already implies that this not being a valid identifier would be a syntax error. But they wanted to give users at least one version where it was a warning, to give an opportunity to see the warning and fix sources before this became a compile error.

FWIW, `buf` already treats this as an error (and we've had had no complaints from users on the discrepancy with `protoc`).